### PR TITLE
Node nvm fallback implement for zsh & bash

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature name**:
 
 * Node
 
 **Description**:
 
 This PR introduces the following functionality :
 
 * Fallback methods implementation for node - nvm installation.
 
 _Changelog_:
 
 * Updated `install.sh`
 * Created `bash_install_node.sh` file - Corresponding installation file for both `bash` & `zsh` execution env.
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

File: src/node/bash_install_node.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (samruddhikhandale) [inline src/node/bash_install_node.sh:1]: Can you rename the file to `src/node/install_nvm_with_retries.sh`?

File: src/node/install.sh [hunk 2]
```diff
@@ -147,46 +147,17 @@ check_packages() {
         rhel)
             if ! rpm -q "$@" > /dev/null 2>&1; then
                 pkg_mgr_update
+                if [ "${ID}" = "centos" ]; then
```
Comment (samruddhikhandale) [inline src/node/install.sh:150]: I think we can remove these changes as the tests are fixed upstream?

File: src/node/install.sh [hunk 3]
```diff
@@ -285,23 +260,17 @@ elif [ "${NODE_VERSION}" = "latest" ]; then
...
-fi
+    set -e
+    umask 0002
+    # Do not update profile - we'll do this manually
+    export PROFILE=/dev/null
+    ./bash_install_node.sh
+    source "${NVM_DIR}/nvm.sh"
+    if [ "${NODE_VERSION}" != "" ]; then
+        nvm alias default "${NODE_VERSION}"
+    fi
```
Comment (samruddhikhandale) [inline src/node/install.sh:273]: Can you remove the indentation here ? The original code seems correct

File: src/node/install.sh [hunk 4]
```diff
@@ -343,6 +313,7 @@ else
     fi
 fi
 
+
```
Comment (samruddhikhandale) [inline src/node/install.sh:316]: remove?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/node/install.sh (hunk 2 above)
Comment (samruddhikhandale) [reply src/node/install.sh:150]: The tests are currently failing.

File: src/node/install.sh (hunk 4 above)
Comment (gauravsaini04) [reply src/node/install.sh]: done

File: src/node/install.sh (hunk 3 above)
Comment (gauravsaini04) [reply src/node/install.sh]: sure

File: src/node/install.sh (hunk 2 above)
Comment (gauravsaini04) [reply src/node/install.sh]: sure

File: src/node/bash_install_node.sh (hunk 1 above)
Comment (gauravsaini04) [reply src/node/bash_install_node.sh]: sure

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Looks good, can you help fix the failing tests?

**Comment by samruddhikhandale** [comment]
Thanks, the changes look complex. Hence, let's hold on to merging this unless we hear more feedback.

**Pull request closed without merging**